pkg/scalers/aws/aws_sigv4.go

```diff
@@ -0,0 +1,88 @@
+package aws
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/amp"
+
+	"github.com/kedacore/keda/v2/pkg/scalers/scalersconfig"
+	httputils "github.com/kedacore/keda/v2/pkg/util"
+)
+
+// Custom round tripper to sign requests
+type roundTripper struct {
+	client *amp.Client
+	region string
+}
+
+var (
+	// ErrAwsAMPNoAwsRegion is returned when "awsRegion" is missing from the config.
+	ErrAwsAMPNoAwsRegion = errors.New("no awsRegion given")
+)
+
+func (rt *roundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	cred, err := rt.client.Options().Credentials.Retrieve(req.Context())
+	if err != nil {
+		return nil, err
+	}
+
+	// This is to avoid the semgrp issue of odd hash.Sum call flow when using hasher.sum(nil)
+	// "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855" is the sha256 of ""
+	const reqHash = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
```

Comment:
I am sorry, I don't have enough time to dive into, but without knowing all the details, I am concerned about this part.

We should add an explanation why is this specific stuff neede, because it is not so obvious from the code. We shouldn't just add stuff to please semgrep imho.

Reply:
> Do we really need to add some constant or do hash on empty object?

We need to sign the request because giving an empty string (not a hashed empty string) fails in the backend as not signed request, so the answer is yes, we need to provide the value (I tested it). 
The question is: Should we calculate the hash from empty or adding a constant with the value? I think that we can place it as constant if we document it well. I mean, adding the constant as magic string without any other info is a problem, but I guess that if we document it there isn't any problem with it. We are already using constants with magic string for other things [like scopes](https://github.com/kedacore/keda/blob/main/pkg/scalers/azure_pipelines_scaler.go#L37)

Reply:
I added as a const magic string with doc of why we're doing.

Reply:
@sguruvar could you please update the comment? we don't need to mention semgrep, but we should add a comment why we need use that hash value etc. Thanks!

Reply:
> @sguruvar could you please update the comment? we don't need to mention semgrep, but we should add a comment why we need use that hash value etc. Thanks!

done